src/tools/compiletest/src/runtest.rs

```diff
@@ -1349,9 +1349,6 @@ impl<'test> TestCx<'test> {
         };
         rustc.arg(input_file);
 
-        // Use a single thread for efficiency and a deterministic error message order
-        rustc.arg("-Zthreads=1");
-
```

Comment:
Question: though if we no longer have the `-Z threads=1`, would *other* non-parallel tests potentially become flaky?

Reply:
The thread of parallel front end is always 1 only if users pass `-Zthreads=n` mannually. On the other hand, the user can easily break the limit by passing the `-Z threads` option(Later option will reset the previous one, we may need to improve on this), so this code doesn't make much sense. 

Defaulting the number of threads to be greater than 1 is intended but will not be implemented for a short time. When we do, I think we still don't need to manually limit `-Z threads=1` in test suit, but change the it to fit the output of a multithreaded environment.